### PR TITLE
Upgrade to esp-idf-5.3.1.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: mbedtls/include/toit_config.h [hunk 1]
```diff
@@ -2645,7 +2627,7 @@
  * The CTR_DRBG generator uses AES-256 by default.
  * To use AES-128 instead, enable \c MBEDTLS_CTR_DRBG_USE_128_BIT_KEY above.
  *
- * AES support can either be achieved through builtin (MBEDTLS_AES_C) or PSA.
+ * AES support can either be achived through builtin (MBEDTLS_AES_C) or PSA.
```
Comment (kasperl) [inline mbedtls/include/toit_config.h:2630]: wat

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: mbedtls/include/default_config.h [hunk 2]
```diff
@@ -1118,7 +1118,7 @@
  *           MBEDTLS_ECP_DP_SECP256R1_ENABLED
  *
  * \warning If SHA-256 is provided only by a PSA driver, you must call
- * psa_crypto_init() before the first handshake (even if
+ * psa_crypto_init() before the first hanshake (even if
```
[the remark this answers is not in the document]
Comment (floitsch) [reply mbedtls/include/default_config.h:1121]: I'm now copying the original file from mbedtls.
It's not worth patching the typos every time, so we keep those now.

File: mbedtls/include/default_config.h [hunk 3]
```diff
@@ -2645,7 +2627,7 @@
  * The CTR_DRBG generator uses AES-256 by default.
  * To use AES-128 instead, enable \c MBEDTLS_CTR_DRBG_USE_128_BIT_KEY above.
  *
- * AES support can either be achieved through builtin (MBEDTLS_AES_C) or PSA.
+ * AES support can either be achived through builtin (MBEDTLS_AES_C) or PSA.
```
[the remark this answers is not in the document]
Comment (floitsch) [reply mbedtls/include/default_config.h:2630]: same: this typo is in the original file, and it's not worth patching it.

File: mbedtls/include/toit_config.h (hunk 1 above)
Comment (floitsch) [reply mbedtls/include/toit_config.h:2630]: same:
I'm taking the original file and only applying the configuration changes that are necessary. It's not worth fixing the typos every time.

**Pull request merged**